src/fields/fft_poisson_solver/fft/WrapDSTW.cpp

```diff
@@ -16,6 +20,18 @@ namespace AnyDST
         const int nx = real_size[0];
         const int ny = real_size[1];
 
+#if defined(AMREX_USE_OMP) && defined(HIPACE_FFTW_OMP)
+        if (nx > 32 && ny > 32) {
+#   ifdef AMREX_USE_FLOAT
+            fftwf_init_threads();
+            fftwf_plan_with_nthreads(omp_get_max_threads());
+#   else
+            fftw_init_threads();
+            fftw_plan_with_nthreads(omp_get_max_threads());
```

Comment:
We could also add, just to expose even more control, a runtime parameter that can overwrite `omp_get_max_threads()` from the inputs file.

The default would be the heuristic you already added (`1` of `<32**2` cells and `omp_get_max_threads()` otherwise), but it could add a useful intermediate layer of control in case we want to set the FFT parallelism independet of the rest of the sum that is controlled by `OMP_NUM_THREADS`.

Reply:
You are right, this will be an interesting addition. After an offline discussion with @MaxThevenet, I will merge this PR as is and add this feature as soon as we have other openMP acceleration. As it is the only function using openMP, we have currently full control with ` OMP_NUM_THREADS`.

Reply:
Oh right, I forgot this is the first OpenMP accelerated part :sweat_smile: